libvast/src/address.cpp

```diff
@@ -21,6 +21,25 @@ address::address() {
   bytes_.fill(0);
 }
 
+address::address(uint8_t const* bytes, family fam, byte_order order) {
+  auto b = reinterpret_cast<const uint32_t*>(bytes);
+  if (fam == ipv4) {
+    std::copy(v4_mapped_prefix.begin(), v4_mapped_prefix.end(), bytes_.begin());
+    auto p = reinterpret_cast<uint32_t*>(&bytes_[12]);
+    if (order == host)
+      *p = detail::to_network_order(*b);
+    else
+      *p = *b;
+  } else if (order == host) {
+    for (auto i = 0u; i < 4u; ++i) {
+      auto p = reinterpret_cast<uint32_t*>(&bytes_[i * 4]);
+      *p = detail::to_network_order(*(b + i));
+    }
+  } else {
+    std::copy(b, b + 4, reinterpret_cast<uint32_t*>(bytes_.data()));
```

Comment:
Casting the input into 2 `uint64_t` operations cuts the operations in half.

(Probably won't matter in practice due to compiler optimizations, but still worthwhile pointing out).